Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/analytics/AnalyticsSettings.kt

```diff
@@ -1,7 +1,35 @@
 package de.rki.coronawarnapp.datadonation.analytics
 
+import android.content.Context
+import com.google.protobuf.ByteString
+import de.rki.coronawarnapp.server.protocols.internal.ppdd.PpaData.ExposureRiskMetadata
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.preferences.FlowPreference
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class AnalyticsSettings @Inject constructor()
+class AnalyticsSettings @Inject constructor(
+    @AppContext private val context: Context
+) {
+    private val prefs by lazy {
+        context.getSharedPreferences("analytics_localdata", Context.MODE_PRIVATE)
+    }
+
+    val previousExposureRiskMetadata = FlowPreference(
```

Comment:
Please add a unit test for the conversion.

Reply:
done 👍